.github/workflows/generate-modified-test-wrappers.yaml

```diff
@@ -0,0 +1,91 @@
+name: generate-wrappers
+
+on:
+  workflow_call:
+    inputs:
+      cli-path:
+        required: false
+        type: string
+      wasm-path:
+        required: false
+        type: string
+
+jobs:
+  generate_wrappers:
+    name: Generate wrappers
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
```

Comment:
no worries, let me explain: the purpose of this job is to be able to interact with both repositories (toolchain & test harness) and be able to build the wrappers (that live in the harness) with the latest changes in the toolchain. (note that this only happens [when one of the packages](https://github.com/polywrap/toolchain/blob/026067f92407fe25044af1077763ba2dfb028981/.github/workflows/js-ci.yaml#L58-L61) necessary to build the wrappers has changed)

answering your question in concrete: yes, we're downloading both repositories in the workflow, being the toolchain [defined in the "toolchain" directory](https://github.com/polywrap/toolchain/blob/026067f92407fe25044af1077763ba2dfb028981/.github/workflows/generate-modified-test-wrappers.yaml#L29) (represented by the path attribute) - you can check [this](https://github.com/actions/checkout/#checkout-multiple-repos-side-by-side) for more info

so, based on this, what we're doing here is:
- download both repositories
- build the toolchain with the latest changes (from the base branch where pr has been opened)
- pass the path of cli & wasm package (in the ci) to wrapper generation in the harness
- when wrappers are built successfully, we update them in the `packages/test-cases/cases/wrappers` folder
- tests will run as they have always do 

the purpose of this is to make sure that when we introduce changes, we make sure our test cases don't break and are updated with the latest changes.

it's also worth mentioning that you will need to create a branch **with the same name** in the test harness, with the modified wrappers (or not), to be able to run the workflow; otherwise, it will fail.

open to any feedback or suggestions, and please feel free to ask anything that needs to be clarified

Reply:
This is fantastic!